### DISCUSSION TITLE
What plugin is used for showing diagnostics?

### DISCUSSION THREAD

**Discussion by Android789515** [LunarVim/LunarVim#4487] [Q&A]
I noticed that I have diagnositcs like errors which look a lot like the plugin from folke called Trouble.  However, I don't see that in my plugin list, so I assume there is another one in use.  I'm not sure how to interact with the feature.

For example, if I have a long error it will go off screen instead of wrapping and I can't see the whole thing.

![image](https://github.com/LunarVim/LunarVim/assets/90219092/a30a57c1-1eb9-49f3-8b1c-7d88377c0e1f)

**Comment by olrtg** [comment]
Based on my knowledge about Neovim I imagine it works something like this: The LSPs attached to the buffer produces diagnostics and Neovim routes those through its diagnostics API. You can read more at `:h diagnostic-api` to see how to interact with that feature.

**Comment by githorse** [comment]
Not an expert, but my understanding is the same as @olrtg's -- that's not a plugin, it's the built-in Neovim LSP engine. (I could be wrong). 

I was also annoyed by the diagnostics not wrapping and just installed Trouble (with the keybindings recommended in the [LunarVim docs](https://www.lunarvim.org/docs/configuration/keybindings#submenu-mapping)). Definitely an improvement.

**Reply by githorse** [reply]
Ok, but actually it still doesn't wrap correctly even in the Trouble quickfix window. Anybody know how to get diagnostics to softwrap consistently so I can always read them?
